src/test/java/io/cdap/plugin/servicenow/sink/ServiceNowSinkConfigTest.java

```diff
@@ -0,0 +1,414 @@
+    Assert.assertEquals("Insert", config.getOperation());
+  }
+
+  @Test
+  public void testCheckCompatibilitySuccess() {
+    MockFailureCollector collector = new MockFailureCollector();
+    ServiceNowSinkConfig config = Mockito.spy(withServiceNowValidationMock(ServiceNowSinkConfigHelper.newConfigBuilder()
+                                                                             .setTableName("tableName")
+                                                                             .build(), collector));
+    Schema actualSchema = Schema.recordOf("actualSchema",
+                                          Schema.Field.of("Id", Schema.of(Schema.Type.INT)),
+                                          Schema.Field.of("StartDate", Schema.nullableOf(Schema.of(Schema.
+                                                                                                     LogicalType.DATE)))
+      ,
+                                          Schema.Field.of("ExtraField", Schema.of(Schema.Type.STRING)),
+                                          Schema.Field.of("Comment", Schema.of(Schema.Type.STRING)));
+
+    Schema providedSchema = Schema.recordOf("providedSchema",
+                                            Schema.Field.of("Id", Schema.of(Schema.Type.INT)),
+                                            Schema.Field.of("StartDate", Schema.nullableOf(Schema.of(Schema
+                                                                                                       .LogicalType.
+                                                                                                       DATE))),
+                                            Schema.Field.of("Comment", Schema.nullableOf(Schema.of
+                                              (Schema.Type.STRING))));
+    config.checkCompatibility(actualSchema, providedSchema, collector);
+    Assert.assertEquals(0, collector.getValidationFailures().size());
+  }
+
+  @Test
+  public void testCheckCompatibilityMissingField() {
+    MockFailureCollector collector = new MockFailureCollector();
+    ServiceNowSinkConfig config = Mockito.spy(withServiceNowValidationMock(ServiceNowSinkConfigHelper.newConfigBuilder()
+                                                                             .setTableName("tableName")
+                                                                             .build(), collector));
+    Schema actualSchema = Schema.recordOf("actualSchema",
+                                          Schema.Field.of("Comment", Schema.of(Schema.Type.STRING)));
+    Schema providedSchema = Schema.recordOf("providedSchema",
+                                            Schema.Field.of("Id", Schema.of(Schema.Type.INT)));
+    try {
+      config.checkCompatibility(actualSchema, providedSchema, collector);
+      Assert.fail("No exception is to be thrown if missing field would be there.");
+    } catch (NullPointerException e) {
```

Comment:
why is NullPointerException thrown? This looks like a bug in the code - we should be adding failures to the collector, but not dereferencing a null pointer

Reply:
yes, there was a bug. Resolved the bug and modified the junit according to that.